core/src/env/test_env.rs

```diff
@@ -417,6 +445,18 @@ impl TestEnvData {
     pub fn dispatch_call(&mut self, call: &[u8]) {
         self.add_dispatched_call(call);
     }
+
+    pub fn call<T: Decode>(
+        &mut self,
+        callee: &[u8],
+        gas: u64,
+        value: &[u8],
+        input_data: &[u8],
+    ) -> Result<T,CallError> {
+        add_call(callee, gas, value, input_data);
+        unimplemented!();
```

Comment:
remove this

Reply:
```suggestion
```

Reply:
I can see why you left it there. The returning of this is a bit more involved.
What I would recommend doing is the following:
Create another field in the test environment data (`Vec<u8>`) that is settable by the user and that is actually taken upon calling `call` to decode into the expected `T`. So upon testing a smart contract using the test environment a tester can completely control it and say what it expects from the outside.

Reply:
I got it!
Also, i need to check error of `call`. Should i add a field of error flag?
Now it can catch only the decoding error, but cant catch error while executing `call`.